.github/workflows/ci-test.yml

```diff
@@ -79,6 +79,55 @@ jobs:
       - name: build image with dev-setup.sh
         run: docker build -f docker/ci/${{ matrix.target_os }}/Dockerfile -t diem/build_environment:test .
 
+  build-images:
+    runs-on: ubuntu-latest-xl
+    continue-on-error: false
+    needs: prepare
+    if: ${{ github.event_name == 'push' && needs.prepare.outputs.test-rust == 'true' }}
+    env:
+      TAG: github-1
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0 #get all the history!!!
+      - id: changes
+        name: determine changes
+        uses: ./.github/actions/changes
+        with:
+          workflow-file: docker-publish.yml
+      - name: Sign in to dockerhub, install image signing cert.
+        uses: ./.github/actions/dockerhub_login
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+          key_material: ${{ secrets.DOCKERHUB_KEY_MATERIAL }}
+          key_name: ${{ secrets.DOCKERHUB_KEY_NAME }}
+          key_password: ${{ secrets.DOCKERHUB_KEY_PASSWORD }}
+      - name: pre build docker images
+        run: |
+          if  [[ ! "$CHANGES_TARGET_BRANCH" =~ "^release-[0-9|.]+$" ]] && [[ ! "$CHANGES_TARGET_BRANCH" =~ "^gha-test-[0-9|.]+$" ]] ; then
```

Comment:
you probably want to build if the target branch is master so you can use the images in cluster test.

Reply:
Since current LBT workflow will be always triggered by push to auto and canary, so I think we actually do not need this step?

Reply:
oh yes, this step is copied from docker-publish workflow which has some other branches we need to filter. Actually we do not need this step here because ci-test and LBT both be triggered by auto and canary by push only. Thx for pointing out

Reply:
Your right, as long as the step is not triggered on prs we're fine.